# Botany 563 Phylogenetic Analysis of Molecular Data (UW-Madison)

A course in the theory and practice of phylogenetic inference from DNA sequence data. Students will learn all the necessary components of state-of-the-art phylogenomic analyses and apply the knowledge to the data analyses of their own organisms.

- Spring 2021: Tuesday and Thursday 1:00-2:15pm (online via Zoom)
- Instructor: Claudia Solis-Lemus, PhD 
    - Email: solislemus@wisc.edu
    - website: https://solislemuslab.github.io/
- Office hours: Tuesday 2:30-3:30pm, or by appointment


## Learning outcomes

By the end of the course, you will be able to

1. Explain in details all the steps in the pipeline for phylogenetic inference and how different data and model choices affect the inference outcomes
2. Plan and produce reproducible scripts with the analysis of your own biological data
3. Justify the data and model choices in your own data analysis
4. Interpret the results of the most widely used phylogenetic methods in biological terms
5. Orally present the results of your own phylogenomic data analyses based on the best scientific and reproducibility practices


## Textbooks

- _Phylogenetics in the Genomic Era_ ([open access book](https://hal.inria.fr/PGE/page/table-of-contents)) by Celine Scornavacca, Frederic Delsuc and Nicolas Galtier (denoted HAL in the schedule)
- _Tree thinking: an introduction to phylogenetic biology_ by David Baum and Stacey Smith (optional: denoted Baum in the schedule)
- _The Phylogenetic Handbook_ by Philippe Lemey, Marco Salemi and Anne-Mieke Vandamme (optional: denoted HB in the schedule)


## Schedule

| Session | Topic | Reading before class | At the end of the session | Notes |
| :---:   | :---: | :---:       | :---:                     | :---: |
| 01/26 | Introduction | [Syllabus](https://github.com/crsl4/phylogenetics-class/blob/master/syllabus.md) | You will know what will be the structure of the class, the learning outcomes and the grading | [lecture1.md](https://github.com/crsl4/phylogenetics-class/tree/master/lecture-notes/lecture1.md) | 
| 01/28 | Motivation: why learning phylogenomics? | HAL 2.1 | You will identify the different components in phylogenomic analyses | |
| 02/02 | Reproducibility crash course | [Notes on mindful programming](https://github.com/crsl4/mindful-programming/blob/master/lecture.md) | You will be able to follow reproducibility and good computing practices throughout the semester (and beyond) | |
| 02/04 | Introduction to sequences | | | |
| 02/09 | Alignment  | | | |
| 02/11
| 02/16 | Orthology detection  | | | |
| 02/18
| 02/23 | Distance and parsimony methods  | | | |
| 02/25
| 03/02 | Models of evolution  | | | |
| 03/04 | Maximum likelihood  | | | |
| 03/09
| 03/11 | Bayesian inference  | | | |
| 03/16
| 03/18
| 03/23 | Model selection  | | | |
| 03/25
| 03/30 | The coalescent  | | | |
| 04/01
| 04/06
| 04/08 | Co-estimation methods  | | | |
| 04/13
| 04/15
| 04/20 | Discussion: Measures of support  | One per group: 1) [Stenz2015](https://pubmed.ncbi.nlm.nih.gov/26117705/), 2) [Lemoine2018](https://www.nature.com/articles/s41586-018-0043-0), 3) [Anisimova2006](https://academic.oup.com/sysbio/article/55/4/539/1675125), 4) [Sayyari2016](https://academic.oup.com/mbe/article/33/7/1654/2579300) | You will be able to compare and contrast the different ways in which we can measure confidence in our phylogenetic estimates | |
| 04/22 | Discussion: Coalescent vs concatenation  | All: HAL 3.4. One per group: 1) [Springer2018](https://www.mdpi.com/2073-4425/9/3/123), 2) [Mendes2018](https://academic.oup.com/sysbio/article/67/1/158/3953674), 3) [Philippe2017](https://europeanjournaloftaxonomy.eu/index.php/ejt/article/view/407), 4) [Springer2016](https://www.sciencedirect.com/science/article/pii/S1055790315002225?via%3Dihub), 5) [Edwards2016](https://www.sciencedirect.com/science/article/pii/S1055790315003309?via%3Dihub) | You will be able to justify the choice of concatenation vs coalescent in specific scenarios | |
| 04/27 | Discussion: Phylogenomics pitfalls  | One per group: 1) [Bravo2019](https://peerj.com/articles/6399/), 2) [Shen2017](https://www.nature.com/articles/s41559-017-0126), 3) [Young2020](https://onlinelibrary.wiley.com/doi/full/10.1111/syen.12406), 4) [Steel2005](https://www.sciencedirect.com/science/article/pii/S0168952505000946?via%3Dihub) | You will be able to describe and analyze some of the main pitfalls of phylogenomic analysis of big data | |
| 04/29 | Discussion: COVID phylogeny  | One per group: 1) [Morel2020](https://academic.oup.com/mbe/advance-article/doi/10.1093/molbev/msaa314/6030946), 2) [Forster2020a](https://www.pnas.org/content/117/17/9241), 3) [Forster2020b](https://www.pnas.org/content/117/23/12524), [Sanchez-Pacheco2020](https://www.pnas.org/content/117/23/12518), [Mavian2020](https://www.pnas.org/content/117/23/12522), 4) [Pipes2020](https://www.biorxiv.org/content/10.1101/2020.06.19.160630v3) | You will be able to assess the strengths and weaknesses of existing estimated phylogenies for COVID19 | |
| 05/04 | Project presentations  | | | |
| 05/06 | Project presentations  | | | |


### More details

See list of topics, grading and academic policies in the [syllabus](https://github.com/crsl4/phylogenetics-class/blob/master/syllabus.md)