# Plant Path 875: Readings in Phylogenomics (UW-Madison)

Spring 2021: Friday 8:55-9:45am (online via Zoom)

**Instructor:** 

- Claudia Solis-Lemus, PhD
- email: solislemus@wisc.edu
- website: https://solislemuslab.github.io/

**Class description:** 
This course will be a journal club where students will read and discuss the current literature related to phylogenomic analyses. The course will focus on topics relevant for biologists using phylogenomics as part of their research. Topics include model misspecifications and violation of assumptions, propagation of alignment errors into the estimation pipeline, and evaluation of the limits of state-of-the-art methods on a variety of real-life datasets.

**Class website:** https://github.com/crsl4/phylogenetics-class/tree/master/PP875

## Topics

We will read papers: 
1. related to weaknesses and limitations in phylogenetic inference:
    - Errors in the alignment/orthology detection
    - Model misspecification
    - Violation of model assumptions e.g.
        - Recombination
        - ILS
        - Gene Flow
        - Presence of Paralogs
        - Molecular clock
2. related to measures of support and confidence values for phylogenetic estimates
3. related to performance evaluations of widely used methods with messy data

## Format of the class

- The class will follow the structure of a jigsaw dynamic:  
    - There will be two groups in class: group 1 and group 2
    - Each group will read one paper ahead of class
- The class format will be:
    - 8:55-9:10 (15min): Each group will be in a breakout room where they will discuss their paper and prepare a 10-min presentation for the whole class
    - 9:10-9:25 (15min): 10-minute presentation of group 1 with 5 minutes for questions
    - 9:25-9:40 (15min): 10-minute presentation of group 2 with 5 minutes for questions
    - 9:40-9:45 (5min): Conclusions

## Schedule

| Session | Reading | Notes |
| :---:   | :---   | :---:                     |
| 01/29   |  Introduction |  |
| 02/05   |  Group 1: Comparison of multiple sequence alignment methods, [link](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0018093) | [slides](https://docs.google.com/presentation/d/1agzopLF6CNrYyQBiyMhkL3ELgBo-3iBZth1-fioZzDM/edit?usp=sharing) |
|         |  Group 2: Alignathon, [link](https://genome.cshlp.org/content/24/12/2077) | |
| 02/12   | Group 1: Do Alignment and Trimming Methods Matter, [link](https://academic.oup.com/sysbio/advance-article/doi/10.1093/sysbio/syaa064/5892776) | [slides](https://docs.google.com/presentation/d/1cOLDFZALCggFCUunLn_OqscF9TKG1M1ubPbSSZkeSsk/edit?usp=sharing) |
|         | Group 2: Evaluating the usefulness of alignment filtering, [link](https://bmcevolbiol.biomedcentral.com/articles/10.1186/s12862-019-1350-2) | |
| 02/19   | Group 1: A New Orthology Assessment Method, [link](https://academic.oup.com/mbe/article/33/8/2117/2578877) | [slides](https://docs.google.com/presentation/d/1r0mx4WA-AOr281h3Zx823LwgjFV8J3nflkxZeSiUOGs/edit?usp=sharing) |
|         | Group 2: Orthology inference: a simulation study, [link](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0056925) | |
| 02/26   | Group 1: How Well Does Your Phylogenetic Model Fit Your Data?, [link](https://academic.oup.com/sysbio/article/68/1/157/5133546) | [slides](https://docs.google.com/presentation/d/1xjY1Wa7T2g80Q5kBEjPYUTYhBllj2kanRxaJxXzEQCo/edit?usp=sharing) |
|         | Group 2: Bayesian selection of misspecified models, [link](https://www.pnas.org/content/115/8/1854) | |
| 03/05   | Group 1: Presence of Paralogs, [link](https://www.sciencedirect.com/science/article/pii/S0168952520302122?via%3Dihub) | [slides](https://docs.google.com/presentation/d/1vqIe0LNpU0msTzVUQSnO2solvZDmTOa_jMb64yN4sUM/edit?usp=sharing) |
|         | Group 2: Molecular Clocks, [link](https://www.sciencedirect.com/science/article/pii/S0168952520301311?via%3Dihub) |  |
| 03/12   | Group 1: Methods to Calculate Concordance Factors, [link](https://academic.oup.com/mbe/article/37/9/2727/5828940) | [slides](https://docs.google.com/presentation/d/1YVKsRL76ZcdoziChZUvSyneFFYnk_1-7qQsDOs5r7XI/edit?usp=sharing) |
|         | Group 2: Novel information theory-based measures, [link](https://academic.oup.com/mbe/article/31/5/1261/994356) | |
| 03/19   | Group 1: When species trees disagree, [link](https://www.biorxiv.org/content/10.1101/2020.03.27.012237v1) | [slides](https://docs.google.com/presentation/d/1v9UkTA0qObrQa1oUOUSRluzmUZ8eg75Zkd5sbCDsufI/edit?usp=sharing) |
|         | Group 2: Assessing Confidence in Root Placement, [link](https://www.biorxiv.org/content/10.1101/2020.07.31.230144v1)
| 03/26   | Group 1: Irreproducibility in maximum likelihood, [link](https://www.nature.com/articles/s41467-020-20005-6) | [slides](https://docs.google.com/presentation/d/1IpRZTz_lv3mN2BxhTuBfclxUFzS3M9mBHF1EK00g_Gk/edit?usp=sharing) |
|         | Group 2: Long branch effects distort maximum likelihood, [link](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0036593) | |
| 04/02   | Group 1: ILS in mammals [link](https://academic.oup.com/sysbio/article/66/1/112/2449707) | [slides](https://docs.google.com/presentation/d/1SSLh4asEhhVHpb78wna1ubNSMARGBbvH_ygnmsZDdI4/edit?usp=sharing) |
|         | Group 2: Sources of error inherent in species-tree estimation, [link](https://academic.oup.com/sysbio/article/59/5/573/1647664) | |
| 04/09   | Group 1: Plant phylogenomics, [link](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5895195/) | [slides](https://docs.google.com/presentation/d/1k69NgMkRtUQk-ZioKs_W3k-nH-0LJwt9OMFKtPfpmRM/edit?usp=sharing) |
|         | Group 2: Why Do Phylogenomic Data Sets Yield Conflicting Trees?, [link](https://academic.oup.com/sysbio/article/66/5/857/3091102) | |
| 04/16   | Group 1: Inferring Ancient Relationships, [link](https://academic.oup.com/icb/article/58/4/623/5049468) | [slides](https://docs.google.com/presentation/d/1CfGru3h2AYQ0QtUwoocla4GZPEA6MBAL6ZoOEJHWMgk/edit?usp=sharing) |
|         | Group 2: Is recombination a problem?, [link](https://academic.oup.com/sysbio/article/61/4/691/1637909) | |
| 04/23   | Group 1: Evaluating Fast Maximum Likelihood, [link](https://academic.oup.com/mbe/article/35/2/486/4644721) | [slides](https://docs.google.com/presentation/d/1Jtyio57XKh0jBL1fGyzJQlgj57P3Uptg7ohhFFflA3M/edit?usp=sharing) |
|         | Group 2: A Critical Review on the Use of Support Values, [link](https://academic.oup.com/mbe/article/34/6/1535/3077051) | |
| 04/30   | Group 1&2: Genome-wide interrogation advances, [link](https://www.nature.com/articles/s41559-016-0020) | [slides](https://docs.google.com/presentation/d/1LxIUbg5i_PPBe5nlQCxtUnpiv7si3ZhF8Cg3ZT4fVz8/edit?usp=sharing) |
|         | Constrained gene tree inference, [link](https://www.nature.com/articles/s41559-016-0056) | |
| 05/07   | Group 1: A new phylogenetic protocol, [link](https://academic.oup.com/nargab/article/2/2/lqaa041/5861483) | [slides](https://docs.google.com/presentation/d/1ieYN8a-UIMIMCoUwQM0_jAFgbb4sgIqWNeAwqk4fR48/edit?usp=sharing) |
|         | Group 2: The effect of alignment uncertainty, [link](https://bmcevolbiol.biomedcentral.com/articles/10.1186/s12862-019-1534-9) | |


